### PR TITLE
[AF-2649] Add params to enable users to set style

### LINKED ISSUE #24: Dynamic Bar Chart: Allow users to set style
Allow users to set bar style and label style using component properties in manifest.json.

### LINKED ISSUE #25: Dynamic Bar Chart: Support Data Transpose
By default dynamic bar chart uses each column as the timeline. Support also dataset transpose, where:

* First column rows contains the timeline valyes (e.g. 2011, 2012 and so soon)
* EAch remaning column is a bar where the rows contains the value for each timeline value;

### PR DESCRIPTION
**Pull request by manaswinidas**
Fixes #24

Fixes #25

### CODE REVIEW

File: bar-chart-race/src/ChartRace.tsx [hunk 1]
```diff
@@ -37,35 +55,68 @@ export class ChartRace extends Component<any, State> {
...
             valuesarray.push(values);
           }
           objectvalues[keyObj] = valuesarray;
         }
         data = objectvalues;
       }
+      data = JSON.parse(JSON.stringify(data));
       this.setState({
-        titleenabled: titleenabled,
+        titleenabled: titleenabled || TITLE_ENABLED,
```
Comment (jesuino) [inline bar-chart-race/src/ChartRace.tsx:107]: This will always resolve to true - AND operation instead OR should work

**Review by jesuino — CHANGES_REQUESTED** [review]
Hello @manaswinidas 

Thank you so much for the great improvements! 

The CSS properties are working very well. The dataset is now columns based, meaning that each column is the BAR, which makes easier to use existing datasets, thanks again!

I noticed a few issues:

1) I see that disabling title property is not working anymore - I click to disable but the title continues showing;
2) I see a lost "undefined" at the end of the bar chart
3) Finally the bars are still not changing position, this may be due some specific issue with the API in my setup, we can discuss later what could be different in my environment.

Besides that, the components looks great, thanks for your work on it! I am waiting for these changes so we can merge this comp.

Thanks!


dynamic_bar

**Comment by manaswinidas** [comment]
Hi @jesuino thanks for reviewing! Sorry for missing the "undefined" earlier :sweat_smile: I have fixed the disable title too. Not sure why the bars aren't changing position on your end, seems to work fine on my end. Will have to have a closer look at it. Might be a potential bug.

**Comment by jesuino** [comment]
Thanks, @manaswinidas  - I am merging this PR to start our API migration

**Review by jesuino — APPROVED** [review]

**Pull request merged**